src/main/java/com/arangodb/internal/net/HostImpl.java

```diff
@@ -41,7 +42,13 @@ public HostImpl(final ConnectionPool connectionPool, final HostDescription descr
 
 	@Override
 	public void close() throws IOException {
-		connectionPool.close();
+		IOUtils.closeQuietly(connectionPool);
+	}
+
+	@Override
+	protected void finalize() throws Throwable {
+		super.finalize();
+		close();
```

Comment:
There is no guarantee finalize method will be invoked ever in Java. So having this update makes little sense. Moreover relying on cleaning resources here is dangerous, since resources will be not disposed on time.

Instead we can make this class implementing Closeable interface so that its users will know that they should close it manually.